src/main/java/bot/Command.java

```diff
@@ -0,0 +1,112 @@
+package bot;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Command class is concerned with parsing the input from the user.
+ */
+public final class Command {
+    /**
+     * The actual command the user inputted.
+     */
+    private final String command;
+
+    /**
+     * This description that came with this command.
+     */
+    private final String desc;
+
+    /**
+     * The extra arguments specified with the command.
+     */
+    private final Map<String, String> arguments;
+
+    /**
+     * Creates a Command object.
+     *
+     * @param command   The actual command inputted by the user.
+     * @param desc      Description for the command.
+     * @param artifacts Extra arguments specified with this command.
+     */
+    private Command(String command, String desc, Map<String, String> artifacts) {
+        this.command = command;
+        this.desc = desc;
+        this.arguments = artifacts;
+    }
+
+    /**
+     * Creates a new Command from a user input line.
+     *
+     * @param input The inputted line from the console.
+     * @return New Command object with parsed information.
+     */
+    public static Command fromString(String input) {
+        // Trim the input.
+        input = input.trim();
+
+        // We first split the command with a space. This will allow us to get the first word
```

Comment:
The comment really help others understanding the methods, although I didn't really get the idea of using HashMap

Reply:
haha the hashmap is so we can store the value of any argument.

this way, we can use the same parser method to parse every single user input, regardless of the command the user wants to type :)

if we require any argument, then we can simply use the key to get the corresponding value, or `null` if it's missing.